### PR TITLE
Configurable max_failures_per_account

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**
### Makes `max_failures_per_account` configurable via a passed in option to producer_plugin.

The default value will remain 3, but will be overwritten if a different number is passed in.  I have also added the new option to the docs.

### CODE REVIEW

File: plugins/producer_plugin/producer_plugin.cpp [hunk 1]
```diff
@@ -1806,7 +1811,10 @@ namespace {
 // track multiple failures on unapplied transactions
 class account_failures {
 public:
-   constexpr static uint32_t max_failures_per_account = 3;
+  explicit account_failures(uint32_t max_failures)
+  :max_failures_per_account(max_failures)
```
Comment (heifner) [inline plugins/producer_plugin/producer_plugin.cpp:1815]: Not sure how this even compiles for you. max_failures_per_account needs to be a member of `account_failures` not `account_failure`

**Review by heifner — CHANGES_REQUESTED** [review]

File: plugins/producer_plugin/producer_plugin.cpp (hunk 1 above)
Comment (ClaytonCalabrese) [reply plugins/producer_plugin/producer_plugin.cpp:1815]: I hadn't had a chance to build it yet. I apologize, it should have been in `draft` to indicate that.

File: plugins/producer_plugin/producer_plugin.cpp [hunk 2]
```diff
@@ -726,6 +727,8 @@ void producer_plugin::set_program_options(
           "Maximum wall-clock time, in milliseconds, spent retiring scheduled transactions in any block before returning to normal transaction processing.")
          ("subjective-cpu-leeway-us", boost::program_options::value<int32_t>()->default_value( config::default_subjective_cpu_leeway_us ),
           "Time in microseconds allowed for a transaction that starts with insufficient CPU quota to complete and cover its CPU usage.")
+         ("subjective-account-max-failures", boost::program_options::value<int32_t>()->default_value(3),
```
Comment (heifner) [inline plugins/producer_plugin/producer_plugin.cpp:730]: Make this an uint32_t to match the variable it is stored in.

File: plugins/producer_plugin/producer_plugin.cpp [hunk 3]
```diff
@@ -726,6 +727,8 @@ void producer_plugin::set_program_options(
...
+          "Sets the maximum amount of failures that are allowed for a given account per block."),
```
Comment (heifner) [inline plugins/producer_plugin/producer_plugin.cpp:731]: The compile error is the `,` at the end of this line.

File: plugins/producer_plugin/producer_plugin.cpp [hunk 4]
```diff
@@ -1806,10 +1811,13 @@ namespace {
 // track multiple failures on unapplied transactions
 class account_failures {
 public:
-   constexpr static uint32_t max_failures_per_account = 3;
 
    //lifetime of sb must outlive account_failures
-   explicit account_failures( const eosio::subjective_billing& sb ) : subjective_billing(sb) {}
+   explicit account_failures( const eosio::subjective_billing& sb, uint32_t max_failures )
+   : subjective_billing(sb),
+     max_failures_per_account(max_failures)
```
Comment (heifner) [inline plugins/producer_plugin/producer_plugin.cpp:1818]: The order here should match the declaration order.

**Review by heifner — APPROVED** [review]

**Review by heifner — APPROVED** [review]

**Pull request merged**